src/main/java/com/thebois/models/beings/Colony.java

```diff
@@ -0,0 +1,45 @@
+package com.thebois.models.beings;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import com.thebois.models.Position;
+
+/**
+ * A Colony is a collection of Pawns that can be controlled by the player.
+ */
+public class Colony implements IBeingGroup {
+
+    private final Collection<IBeing> pawns = new ArrayList<>();
+
+    @Override
+    public Collection<IBeing> getBeings() {
+        return pawns;
+    }
+
+    @Override
+    public void createBeing() {
+        final Position spawnPosition = new Position();
+        pawns.add(new Pawn(spawnPosition));
+    }
+
+    @Override
+    public void createBeing(Position spawnPosition) {
+        pawns.add(new Pawn(spawnPosition));
+    }
+
+    @Override
+    public void createBeing(Position spawnPosition, int numberOfNewBeings) {
+        for (int i = 0; i < numberOfNewBeings; i++) {
+            pawns.add(new Pawn(spawnPosition));
+        }
+    }
```

Comment:
Does this mean that there should be a method for adding beings to a beingGroup? Where do "unclaimed" beings exist? i.e. ones created by a house or something akin.

Reply:
No, I don't think some outside client should be able to *tell* the `BeingGroup` to take in a new `Being`. 

I'm thinking that there's no such thing as an "unclaimed" being. Beings are only created by `BeingGroups` and they only belong to the group that created them (unless we implement kidnapping :P).

When a house is placed, the `BeingGroup` that is owned by the player (the Colony), can listen to that in the form of an observer, and when it gets an event indicating that a `House` has been placed, it will create a new `Being` for the house.

What do you think of this way of doing it?

Reply:
Do you think a viable solution would be to pass the BeingGroup to the Structure being built, and then the Structure passing itself to the BeingGroup when it is fully built, in order to notify the BeingGroup? So that the BeingGroup can decide what to do, i.e. create new Beings or something else. 

This would be an implementation of the Visitor Design Pattern.

Reply:
I don't think it's a good idea to have `Structure` depend on `BeingGroup`.

No need to use the visitor pattern, I'd say observer is enough here:

`World` is the observable, generating "structure created" events.

`BeingGroup` is the observer, listening to the structure events from `World`.
When `BeingGroup` recieves an event, it gets to know what has been built and can react in any way it wants to. This lets `BeingGroup` decide what to do, as you suggested.